### PR TITLE
Restore Channels ReadAsync implementation

### PR DESCRIPTION
**Pull request by tarekgh**

### CODE REVIEW

**Comment by tarekgh** [comment]
@stephentoub I have brought back the implementation of ReadSync on all Channels classes. one question, do we really need to use AutoResetAwaiter? I have brought it because it was used but I am not sure if it is really needed.

**Comment by tarekgh** [comment]
cc @joshfree

**Comment by stephentoub** [comment]
> do we really need to use AutoResetAwaiter?

Nope. It was there for GetAwaiter, which was removed.

**Comment by tarekgh** [comment]
good, I'll get rid of AutoResetAwaiter then.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by tarekgh** [comment]
@stephentoub The code coverage is almost 100% in all areas except 

```
System.Threading.Channels.ChannelReader`1    32.1%    14.2% 
```

And that is expected as this code is not consumed by any of our concrete channels. I'll try to write a test force exercising this code too so we'll go back to close to 100% 
I don't see any old/removed tests can help in this scenario.

**Comment by stephentoub** [comment]
Ok. Thanks, @tarekgh.

File: src/System.Threading.Channels/tests/ChannelTests.cs [hunk 1]
```diff
@@ -131,6 +131,72 @@ public void DefaultCompletion_NeverCompletes()
...
         }
 
+        [Fact]
+        public async void TestBaseClassReadAsync()
+        {
+            WrapperChannel<int> channel = new WrapperChannel<int>(10);
+            var reader = channel.Reader;
+            var writer = channel.Writer;
+
+            // 1- do it through synchoronouse TryRead()
```
Comment (stephentoub) [inline src/System.Threading.Channels/tests/ChannelTests.cs]: Nit: spelling of synchronous

File: src/System.Threading.Channels/tests/ChannelTests.cs [hunk 2]
```diff
@@ -131,6 +131,72 @@ public void DefaultCompletion_NeverCompletes()
             await Assert.ThrowsAsync<FieldAccessException>(() => t);
         }
 
+        [Fact]
+        public async void TestBaseClassReadAsync()
+        {
+            WrapperChannel<int> channel = new WrapperChannel<int>(10);
+            var reader = channel.Reader;
+            var writer = channel.Writer;
```
Comment (stephentoub) [inline src/System.Threading.Channels/tests/ChannelTests.cs]: Nit: can you use the actual type names instead of var here?

**Comment by tarekgh** [comment]
@MattGal is this known issue? 

https://ci3.dot.net/job/dotnet_corefx/job/master/job/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/8265/

```
  System.IO.FileSystem.Watcher.Tests -> /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/bin/OSX.AnyCPU.Debug/System.IO.FileSystem.Watcher.Tests/netstandard/System.IO.FileSystem.Watcher.Tests.dll
  System.Net.HttpListener.Tests -> /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/bin/OSX.AnyCPU.Debug/System.Net.HttpListener.Tests/netstandard/System.Net.HttpListener.Tests.dll
    File "/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/Tools/dumpling.py", line 1
      <!DOCTYPE html>
      ^
  SyntaxError: invalid syntax
  /usr/bin/python: can't open file '/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/tempHome/.dumpling/dumpling.py': [Errno 2] No such file or directory
  
  
  /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/dir.traversal.targets(81,3): warning MSB4011: "/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/Tools/Dumpling.targets" cannot be imported again. It was already imported at "/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/Tools/tests.targets (558,3)". This is most likely a build authoring error. This subsequent import will be ignored. [/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/src/tests.builds]
```

I am merging this change and we can log an issue if this is unknown

**Comment by MattGal** [comment]
@tarekgh this was addressed by https://github.com/dotnet/buildtools/pull/1900 , and is a symptom of the dumpling server being offline.

**Pull request merged**